android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -330,11 +333,11 @@ public void onError(@NonNull CameraDevice cameraDevice, int errorCode) {
         null);
   }
 
-  private void writeToFile(ByteBuffer buffer, File file) throws IOException {
+  private void writeToFile(Bitmap bitmap, File file) throws IOException {
     try (FileOutputStream outputStream = new FileOutputStream(file)) {
-      while (0 < buffer.remaining()) {
-        outputStream.getChannel().write(buffer);
-      }
+      bitmap.compress(Bitmap.CompressFormat.JPEG, 90, outputStream);
```

Comment:
O šita vieta kodėl keista? Turim mes compressinimą dart pusėje, kur useris gali parametrus nustatyti. Tai kaip čia dabar bus - du kartus kompresins su JPEG 90% ?

Reply:
Nes po to nuotraukos pavertimo mes nebeturim ByteBuffer kurio viduje jau jpeg nuotrauka yra, o bitmap, su savo vidiniu formatu. Su tuo compress metodu realiai paprasčiausias būdas bitmap'ą paversti į jpeg formatą, galėčiau čia turbūt 100% quality palikti, arba kažkaip pakeisti dart kode compressinimą.

Reply:
O kaip dažnai tas pavertimas daromas? Čia matyt klausimas, ar visada reikia compressinti, gal čia retas atvejis? Jei nedažnai, tai gal tada būtų galima palikti originalų kodą, kai `getMediaOrientation() == 0` - neprarastų nuotrauka kokybės ir nereiktų dart code keist compressinimo logikos

Reply:
Man atrodo, kad dažniausiai kameros sensorius yra pasuktas taip, kad jo natūralus nulinis kampas yra kai fotografuoji landscape, čia bent jau mano pastebėjimas iš device su kuriais testavau. O šeip tai tas pavertimas gali būti reikalingas bet kada, `getMediaOrientation` naudoja ir dabartinį momentinį device orientation. Ar labai pasijaus per nuotraukos kokybę čia jeigu su 100 quality konvertuosiu į jpeg čia? Žinau kad jpeg konvertavimas visada yra lossy, bet gal ne tiek daug :)

Reply:
Nu 100 gal čia bus sensible default - tada dart kode nereiks keisti.

Reply:
Pakeičiau.